src/Sarif.UnitTests/TestData/InsertOptionalDataVisitor/CoreTests_FlattenedMessages.sarif

```diff
@@ -11,6 +11,32 @@
         "sarifLoggerVersion": "1.5.22.0",
         "language": "en-US"
       },
+      "invocations": [
+        {
+          "toolNotifications": [
+            {
+              "message": {
+                "text": "A notification was raised. Notification kind was: 'tool notification'.",
```

Comment:
>text [](start = 17, length = 4)

Yup. And here, when you _do_ set that flag, it _does_ flatten the notifications.